hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenMintHandler.java

```diff
@@ -239,4 +264,53 @@ private Nft buildNewlyMintedNft(
                 .metadata(meta)
                 .build();
     }
+
+    @NonNull
+    @Override
+    public Fees calculateFees(@NonNull final FeeContext feeContext) {
+        final var op = feeContext.body().tokenMintOrThrow();
+        final var readableTokenStore = feeContext.readableStore(ReadableTokenStore.class);
+        final var tokenType = readableTokenStore.get(op.tokenOrThrow()).tokenType();
+        final var subType = tokenType == TokenType.FUNGIBLE_COMMON
+                ? SubType.TOKEN_FUNGIBLE_COMMON
+                : SubType.TOKEN_NON_FUNGIBLE_UNIQUE;
+
+        final var readableAccountStore = feeContext.readableStore(ReadableAccountStore.class);
+        final var payerId = feeContext.body().transactionIDOrThrow().accountIDOrThrow();
+        final var payerKey = readableAccountStore.getAccountById(payerId).keyOrThrow();
+
+        final var calculator = feeContext.feeCalculator(subType);
+        if (SubType.TOKEN_NON_FUNGIBLE_UNIQUE.equals(subType)) {
+            // The price of nft mint should be increased based on number of signatures.
+            // The first signature is free and is accounted in the base price, so we only need to add
+            // the price of the rest of the signatures.
+            calculator.addVerificationsPerTransaction(Math.max(0, numSimpleKeys(payerKey) - 1L));
+        }
+
+        final var meta = TOKEN_OPS_USAGE_UTILS.tokenMintUsageFrom(fromPbj(feeContext.body()), fromPbj(subType), 76000L);
```

Comment:
What's this 7600L number? Please add comments here.

Reply:
yes , that parameter is not used. So added a default value. Will add comments

Reply:
Added comment